### PR TITLE
Persistent space collapsed states

### PR DESCRIPTION
**Pull request by SimonBrandner**
Fixes [#17164](https://github.com/vector-im/element-web/issues/17164)

https://user-images.githubusercontent.com/25768714/117127601-93bbe480-ad9c-11eb-93b1-32037fb6157d.mp4

### CODE REVIEW

**Comment by SimonBrandner** [comment]
I didn't use a store because it felt unnecessary for the few lines this has, but sure I can make it into one

File: src/components/views/spaces/SpaceTreeLevel.tsx [hunk 1]
```diff
@@ -68,8 +69,15 @@ export class SpaceItem extends React.PureComponent<IItemProps, IItemState> {
...
 
+        const collapsed = SpaceTreeLevelLayoutStore.instance.getSpaceCollapsedState(
+            props.space.roomId,
+            this.props.parents,
+            !props.isNested,
+        );
+
         this.state = {
-            collapsed: !props.isNested, // default to collapsed for root items
+            // default to collapsed for root items
```
Comment (turt2live) [inline src/components/views/spaces/SpaceTreeLevel.tsx:79]: this comment doesn't make much sense anymore given we're not doing anything to the variable ourselves. I'd suggest moving it inline to Line 75 where we set the default.

**Review by turt2live — APPROVED** [review]
overall looks great, thank you :) 

Just a small comment about a comment location then this is fine to merge.

**Comment by turt2live** [comment]
> I didn't use a store because it felt unnecessary for the few lines this has

It is indeed quite small, but that's okay. The important thing with stores (for us) is that they help disassociate functional behaviour away from the components themselves. It is technically possible to put a dedicated skin over the react-sdk (in most places, but eventually everywhere) and it'd be annoying for the skin-writer to have to copy/paste a bunch of the component. The store isn't perfect as a solution, but it does mean that there's a level of backwards compatibility if the upstream project decided to skin the component and didn't want to lose the user's settings.

**Review by turt2live — APPROVED** [review]

**Pull request merged**